protocol/tx/transaction.go

```diff
@@ -2,15 +2,20 @@ package tx
 
 import (
 	"chain/crypto/sha3pool"
+	"chain/errors"
 	"chain/protocol/bc"
 	"fmt"
 )
 
+func init() {
+	bc.TxHashesFunc = TxHashes
+}
```

Comment:
not necessarily now, but would it possible be able to invert the dependency and make `protocol/bc` depend on `protocol/tx`? This init feels a little fragile (and requires all of those `init` funcs in our tests). We'd need to export more of the `tx` package's types and move the `bc.Tx`-to-graph conversion into `bc` but neither seems bad?